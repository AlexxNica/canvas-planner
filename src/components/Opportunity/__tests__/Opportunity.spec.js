import React from 'react';
import { shallow } from 'enzyme';
import { Opportunity } from '../index';

const defaultProps = {
  dueAt: "2017-03-09T20:40:35Z",
  courseName: "course about stuff",
  opportunityTitle: "this is a description about the opportunity",
  points: 20,
  url: "http://www.non_default_url.com",
  timeZone: 'America/Denver',
};

it('renders the base component correctly', () => {
  const wrapper = shallow(
    <Opportunity {...defaultProps} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders the base component correctly without points', () => {
  let tempProps = defaultProps;
  tempProps.points = null;
  const wrapper = shallow(
    <Opportunity {...tempProps} />
  );
  expect(wrapper).toMatchSnapshot();
});
