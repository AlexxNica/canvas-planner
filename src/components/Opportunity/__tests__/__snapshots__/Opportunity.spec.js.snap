// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the base component correctly 1`] = `
<div>
  <header>
    <div>
      course about stuff
    </div>
    <div>
      <Button
        as="button"
        buttonRef={[Function]}
        fluidWidth={false}
        margin="0"
        size="small"
        type="button"
        variant="icon"
      >
        <IconXLine
          title="Close"
        />
      </Button>
    </div>
  </header>
  <Container
    as="div"
    display={null}
    margin="0 0 small"
  >
    <Link
      as="button"
      ellipsis={false}
      href="http://www.non_default_url.com"
      linkRef={[Function]}
    >
      this is a description about the opportunity
    </Link>
  </Container>
  <footer>
    <div>
      <Pill
        text="Badge list..."
        variant="default"
      />
      <div>
        <span>
          Due:
        </span>
         
        March 9, 2017 at 1:40 PM
      </div>
    </div>
    <div>
      <span>
        20
      </span>
      points
    </div>
  </footer>
</div>
`;

exports[`renders the base component correctly without points 1`] = `
<div>
  <header>
    <div>
      course about stuff
    </div>
    <div>
      <Button
        as="button"
        buttonRef={[Function]}
        fluidWidth={false}
        margin="0"
        size="small"
        type="button"
        variant="icon"
      >
        <IconXLine
          title="Close"
        />
      </Button>
    </div>
  </header>
  <Container
    as="div"
    display={null}
    margin="0 0 small"
  >
    <Link
      as="button"
      ellipsis={false}
      href="http://www.non_default_url.com"
      linkRef={[Function]}
    >
      this is a description about the opportunity
    </Link>
  </Container>
  <footer>
    <div>
      <Pill
        text="Badge list..."
        variant="default"
      />
      <div>
        <span>
          Due:
        </span>
         
        March 9, 2017 at 1:40 PM
      </div>
    </div>
    <div>
      <span />
      points
    </div>
  </footer>
</div>
`;
